### PR TITLE
Capt

### PR DESCRIPTION
**Pull request by hsinkoff**

### CODE REVIEW

**Comment by washburnad** [comment]
I have a couple of thoughts/comments.

1. It looks like capture is called on an instance of game.  My initial thought is that it needs to return a value (true or false) to let the moving piece know that a piece was captured.  However, I don't think it actually matters.  
2. Capture logic does need to take into account color of piece that is moving and color of piece being captured.  Perhaps ```def capture( x, y, color )``` and add a conditional to check that colors are different.
3. Can reduce 3 lines to one with c.update_attributes(...., ....., .....).  I think it's also fewer queries to the db.  This could also be a piece.rb method that gets called on the piece as a refactor, keeping piece modifications in piece.rb

**Comment by kenmazaika** [comment]
this is good feedback, aaron.

**Comment by hsinkoff** [comment]
Aaron,
  You're right the capture branch is being abandoned.  The capt branch however, will need a merge at some point.
  I had forgotten to account for color, but will go back and do so. However, I'm thinking that will also affect the obstructed methods.  As it is now any piece that is landed on is captured, but once I change it we need to say that a piece that is landed on but the same color causes an obstruction.  I believe it just means updating the < or > signs to include equal in the obstruction method within the if right above the while.

**Comment by johnellison** [comment]
@hsinkoff are you talking about the `game.obstruction?` method? or the individual `obstructed_move?` methods?

The `game.obstruction` method is really just determining if there is a piece at a given x and y position. I'm wondering whether a color-agnostic method to determine whether there is a piece at a given location on the game-board is in fact useful. That way we implement the color-specific check within the capture logic.

**Comment by hsinkoff** [comment]
@johnellison I was thinking of the obstructed_move?

**Comment by johnellison** [comment]
I just realized after I wrote the comment and tried to implement it on the Pawn that we do need to have the color of the piece causing the `obstruction` in order to be able to pass the color parameter into the `capture_move?` method to determine whether it is valid or not...

**Comment by washburnad** [comment]
I just added a (flow chart in Trello) [https://trello.com/c/gc2DzNAi/42-create-flow-chart-for-implementing-move-logic-methods] with a comment that addresses the relationship between capture and obstructed_move?

File: app/models/game.rb [hunk 1]
```diff
@@ -57,9 +57,23 @@ def initialize_board!
...
+  def capture(x, y)
+    #check if a piece exists at this coordinate
+    c=self.pieces.where(x_position:x, y_position:y).last
+    #if nil do nothing-- if piece, mark as captured and move to (nil, nil)
+    if c == nil
+      return
+    else
+      c.update_attribute(:captured?, true)
+      c.update_attribute(:x_position, nil)
+      c.update_attribute(:y_position, nil)
```
Comment (washburnad) [inline app/models/game.rb]: Do we want to move these three lines to a piece.rb method (perhaps captured() or mark_captured())?  It seems the query logic is going to be taken care of in the obstructed_move? method.

**Comment by hsinkoff** [comment]
Aaron, I think we probably do want to move the update portion and then we can use it within the obstructed_move.  Once we have obstructed_move figured out with the capture included for one piece, I think we close this PR without merging.

**Comment by johnellison** [comment]
Are we waiting on accepting / closing this PR?

**Pull request closed without merging**